libr/core/cmd_search.c

```diff
@@ -1600,8 +1611,18 @@ static void do_esil_search(RCore *core, struct search_parameters *param, const c
 	const bool cfgDebug = r_config_get_i (core->config, "cfg.debug");
 	RSearch *search = core->search;
 	RSearchKeyword kw = R_EMPTY;
-	if (input[0] == 'E' && input[1] != ' ') {
-		eprintf ("Usage: /E [esil-expr]\n");
+	if (input[0] != 'E') {
+		return;
+	}
+	PJ *pj = NULL;
+	if (input[1] == 'j') { // "/Ej"
+		// BUGGY and dupe not using pj, param->outmode = R_MODE_JSON;
```

Comment:
what does this line mean exactly?

Reply:
it means that cbHit() is also handling the JSON mode when printing...but:

* its not using the pj api
* the output is not valid json
* the output have json escaping issues

so i decided to use local pj for this case because when we do /Ej i dont expect flags to be created, but just a json to be printed. further discussion must be done for the other uses of this function

Reply:
It's better to open a separate issue for this btw.